product-development/README.md

```diff
@@ -0,0 +1,89 @@
+# _**The content in this page applies starting on 2020 March 30**_
+There are no changes to how we work with Google docs and Clubhouse stories prior to this date.
+
+# Product Development
+
+## Workflow
+We have a lightweight product develoment workflow that empowers individual Tilters to collaboratively solve business problems, work directly with customers, and to be accountable for achieving positive business outcomes.
+
+<img src="images/workflow.png" height="300" />
+
+The workflow is summarized in the image above. At the far left, we continually have conversations with customers and the community. These are synthesized into GitHub issues and Clubhouse stories. Based on this feedback and our broader company goals and strategy, we maintain a backlog of high-level initiatives as Clubhouse epics (primarily product changes). Over time, epics are further defined with scoped work and acceptance criteria, as they are eventually implemented and shipped.
+
+### Free-form discussion in Clubhouse stories
+All Tilters are encouraged to collaborate in _discussing_ general ideas in a free-form fashion, with no strict process, no structure, no specific timelines, and no specific implementation commitments. Tilters should prefer discussing ideas in Clubhouse stories, linking ideas together, thus building a web of good ideas that are easily searchable for posterity.
+
+### Defining high-level business initiatives with Clubhouse epics
+All Tilters are encouraged to collaborate in _definining_ high-level business initiatives. Similar to general ideas, there is no strict process for collaboration, except that each business initiative should be documented as a Clubhouse epic. As an epic becomes increasingly well-defined over time, it should have more and more of these components with increasing fidelity:
+- Problem statement
+- Solution scope
+- Attached stories with relevant ideas, including designs
+
+Victor is responsible for ensuring that the [backlog of unstarted epics](https://app.clubhouse.io/windmill/epics?state_ids=500008016&state_ids=500008001) is continually full and that epics are increasingly being better defined. There should be no shortage of epics. Dan B and Nick and Victor are responsible for maintaining the priority of the backlog (higher priority toward the top in the Clubhouse view, when the triangle is pointing down).
+
+<img src="images/epics-priority.png" height="100" />
+
+## Picking the highest priority epic
+When a Tilt engineer is free, they should pick the highest priority unstarted epic to work on and make themself the _sole_ owner of the epic. This indicates that they are the [DRI](https://medium.com/@mmamet/directly-responsible-individuals-f5009f465da4) for the epic. The engineer may find other folks to collaborate on the epic as needed, but they are the DRI for driving the epic to completion. There should be at most one owner on any epic, and that owner is the DRI.
```

Comment:
Prior to the teams approach devs would:
* all be working on their own tickets in silos
* not know what anyone else was working on
* feel like they weren't really collaborating with coworkers

As a dev, I'd often receive a complicated PR from another dev for some feature I'd never heard of, and have no idea problem it was solving, and it'd make it hard and expensive to review. Subteams allowed us to ensure we had a clear set of people who'd be able to review our code and/or help out on tasks.

So, a concrete suggestion and a vague concern, that I'm still processing:

Suggestion: when a dev takes an epic, they find another engineer to be their PR friend
Concern: we're all going to end up reverting to our natural siloed state, working on 4-5 separate projects

Reply:
> So, a concrete suggestion 

I don't want to make it a rule that you must always form a team of `n` people in order to take on an epic. I updated the language so that it encourages people to work in teams, but at their discretion, and also including your reasons. Take a look and see if it makes sense.

> Concern: we're all going to end up reverting to our natural siloed state, working on 4-5 separate projects

Since the epics are intended to be small and finished within 7 business days, and that as part of the acceptance criteria they should be presented at least at the weekly company meeting, I expect:

- At any given moment, there could be 3, or 4 in progress epics
- You would likely be working with at least one other person on at least one of your current epics.
- You would hear about other epics when they are being presented at the company meeting. And even though they would be done, you would be able to suggest follow-up improvements, and be excited to participate in those (since the epics are designed to be small).

I'm expecting that the weekly demos/presentations should be a great catalyst to help people see what others are doing and get excited. What do you think?